crates/egui_demo_lib/src/demo/sliders.rs

```diff
@@ -132,23 +137,43 @@ impl super::View for Sliders {
                 .logarithmic(true)
                 .smart_aim(*smart_aim)
                 .text("left")
-                .trailing_fill(*trailing_fill),
+                .trailing_fill(*trailing_fill)
+                .handle_shape(*handle_shape),
         );
         ui.add(
             Slider::new(max, type_min..=type_max)
                 .logarithmic(true)
                 .smart_aim(*smart_aim)
                 .text("right")
-                .trailing_fill(*trailing_fill),
+                .trailing_fill(*trailing_fill)
+                .handle_shape(*handle_shape),
         );
 
         ui.separator();
 
         ui.checkbox(trailing_fill, "Toggle trailing color");
-        ui.label("When enabled, trailing color will be painted up until the circle.");
+        ui.label("When enabled, trailing color will be painted up until the handle.");
 
         ui.separator();
 
+        ui.label("Handle shape");
+        ui.horizontal(|ui| {
+            ui.radio_value(handle_shape, HandleShape::Circle, "Circle");
+            if ui
+                .radio(
+                    matches!(handle_shape, HandleShape::Rect { aspect_ratio: _ }),
+                    "Rectangle",
+                )
+                .clicked()
+            {
+                *handle_shape = HandleShape::Rect { aspect_ratio: 0.5 };
+            }
+            if let HandleShape::Rect { aspect_ratio } = handle_shape {
+                ui.add(Slider::new(aspect_ratio, 0.1..=3.0).text("Aspect ratio"));
+            }
```

Comment:
Duplicated code - I suggest you create an `impl HandleSape { fn ui(&mut self, ui: &mut Ui) { … } }` for this in `style.rs`